crypto/sha/asm/keccak1600-ppc64.pl

```diff
@@ -668,7 +668,7 @@
 	subi	$out,r4,1		; prepare for stbu
 	mr	$len,r5
 	mr	$bsz,r6
-	${UCMP}i r7,0                   ; r7 = 'next' argument
+	cmplwi	r7,0                    ; r7 = 'next' argument
```

Comment:
Looks ok to me.. Took me a while to understand this instruction...

cmplwi = compare logical word immediate (does an unsigned compare of a 32 bit register with a 16 bit immediate value that is zero extended)
cmpldi = compare logical double word immediate
cmpwi = signed compare

(cmpldi rA, UI maps to cmpli cr0, 0, rA, u16)

Reply:
For PPC64 w = 32bit d = 64bit. And that is the intention - the next argument is int -> 32 bit.